From: Coly Li <bosong.ly@taobao.com>
Date: Fri Jan 20 18:03:32 CST 2012
Subject: [PATCH] power management: disable pcie_aspm on default
Patch-mainline: in house patch
Reference:

On supermicro greencomputing servers, we observe frequently NIC
offline. The issue is caused by hardware compatibility. After
talking with pcie_aspm patch author, this feature is designed
for mobile or portable device, which has negtive performance
impect to network response and throughput.

This patch disables PCIE_ASPM (active state power management)
on default to get better pcie device performance and make a
workaround for greencomputing server NIC offline issue.

Signed-off-by: Coly Li <bosong.ly@taobao.com>
---
Index: linux-2.6.32-131.21.1.el5/drivers/pci/pcie/aspm.c
===================================================================
--- linux-2.6.32-131.21.1.el5.orig/drivers/pci/pcie/aspm.c	2012-01-31 16:40:39.257993843 +0800
+++ linux-2.6.32-131.21.1.el5/drivers/pci/pcie/aspm.c	2012-01-31 16:40:43.242013599 +0800
@@ -75,7 +75,7 @@
 #define POLICY_DEFAULT 0	/* BIOS default setting */
 #define POLICY_PERFORMANCE 1	/* high performance */
 #define POLICY_POWERSAVE 2	/* high power saving */
-static int aspm_policy = POLICY_POWERSAVE;
+static int aspm_policy = POLICY_PERFORMANCE;
 static const char *policy_str[] = {
 	[POLICY_DEFAULT] = "default",
 	[POLICY_PERFORMANCE] = "performance",
